core/src/serve_repair.rs

```diff
@@ -727,6 +697,60 @@ impl ServeRepair {
         }
     }
 
+    fn requires_ping_check(
+        request: &RepairProtocol,
+        root_bank: &Bank,
+        check_ping_ancestor_request_epoch: Option<Epoch>,
+    ) -> bool {
+        match request {
+            RepairProtocol::LegacyWindowIndex(_, _, _)
+            | RepairProtocol::LegacyHighestWindowIndex(_, _, _)
+            | RepairProtocol::LegacyOrphan(_, _)
+            | RepairProtocol::LegacyWindowIndexWithNonce(_, _, _, _)
+            | RepairProtocol::LegacyHighestWindowIndexWithNonce(_, _, _, _)
+            | RepairProtocol::LegacyOrphanWithNonce(_, _, _)
+            | RepairProtocol::LegacyAncestorHashes(_, _, _)
+            | RepairProtocol::Pong(_) => false,
+            RepairProtocol::WindowIndex { .. }
+            | RepairProtocol::HighestWindowIndex { .. }
+            | RepairProtocol::Orphan { .. } => true,
+            RepairProtocol::AncestorHashes { slot, .. } => {
+                Self::should_check_ping_ancestor_request(
+                    *slot,
+                    root_bank,
+                    check_ping_ancestor_request_epoch,
+                )
+            }
+        }
+    }
+
+    fn request_to_ping_packet(
+        request: &RepairProtocol,
+        dest_addr: &SocketAddr,
+        ping: Ping,
+    ) -> Option<Packet> {
```

Comment:
This is a very confusing function.
If you don't need a ping for e.g. `LegacyWindowIndex` why already create a `ping: Ping` and pass it to this function?!

Reply:
restructured to collapse `match` statements